### PR TITLE
[Complications] Rename Provider terminology to DataSource

### PR DESCRIPTION
**Pull request by alexvanyo**
The terminology for complications has changes from `Provider` to `DataSource`. This updates to use the new API, and also renames everything (including the sample itself) to follow the new terminology.

### CODE REVIEW

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/LongTextDataSourceService.kt [hunk 1]
```diff
@@ -19,31 +19,32 @@ import android.app.PendingIntent
...
  * If you don't perform any suspending operations to update your complications, you can subclass
- * [ComplicationProviderService] and override [onComplicationRequest] directly.
- * (see [NoDataProviderService] for an example)
+ * [ComplicationDataSourceService] and override [onComplicationRequest] directly.
+ * (see [NoDataDataSourceService] for an example)
  */
-class LongTextProviderService : SuspendingComplicationProviderService() {
+class LongTextDataSourceService : SuspendingComplicationDataSourceService() {
     override suspend fun onComplicationRequest(request: ComplicationRequest): ComplicationData? {
+        ComplicationDataSourceService
```
Comment (jdkoren) [inline WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/LongTextDataSourceService.kt:47]: Not sure what this line is for

Comment (ADClarke) [reply WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/LongTextDataSourceService.kt:47]: +1 does this compile?

**Review by ADClarke — CHANGES_REQUESTED** [review]
Mostly looks fine bar the one issue.

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/LongTextDataSourceService.kt (hunk 1 above)
Comment (alexvanyo) [reply WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/LongTextDataSourceService.kt]: It does compile, funnily enough! (it's just a noop reference to a companion object)

Definitely a mistake, I think I was adding it to get the import for the KDoc.

Removed in 0d5dffd

**Review by ADClarke — APPROVED** [review]
Thanks!

**Pull request merged**